qbraid/transpiler/conversions/openqasm3/openqasm3_to_cudaq.py

```diff
@@ -0,0 +1,188 @@
+# Copyright (C) 2024 qBraid
+#
+# This file is part of the qBraid-SDK
+#
+# The qBraid-SDK is free software released under the GNU General Public License v3
+# or later. You can redistribute and/or modify it under the terms of the GPL v3.
+# See the LICENSE file in the project root or <https://www.gnu.org/licenses/gpl-3.0.html>.
+#
+# THERE IS NO WARRANTY for the qBraid-SDK, as per Section 15 of the GPL v3.
+
+"""
+Module containing OpenQASM to CUDA-Q conversion function
+
+"""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Dict, Union
+
+import pyqasm
+from openqasm3 import ast
+from qbraid_core._import import LazyLoader
+
+from qbraid.transpiler.annotations import requires_extras, weight
+from qbraid.transpiler.exceptions import ProgramConversionError
+
+cudaq = LazyLoader("cudaq", globals(), "cudaq")
+
+if TYPE_CHECKING:
+    from cudaq import PyKernel, QuakeValue
+
+    from qbraid.programs.typer import QasmStringType
+
+
+def make_gate_kernel(name: str, targs: tuple[type]) -> PyKernel:
+    """Returns CUDA-Q kernel for pure standard gates (no modifiers - ctrl or adj)."""
+
+    if name in [
+        "x",
+        "y",
+        "z",
+        "rx",
+        "ry",
+        "rz",
+        "h",
+        "s",
+        "t",
+        "sdg",
+        "tdg",
+        "u3",
+        "i",
+        "id",
+        "iden",
```

Comment:
I think if we have `i` and `id` that's good. Not sure `iden` is as common of a convention, unless that's used by cudaq. lmk